android/src/main/java/com/geektime/rnonesignalandroid/RNOneSignal.java

```diff
@@ -49,821 +49,546 @@ of this software and associated documentation files (the "Software"), to deal
+    }
+
+    @ReactMethod
+    public void removeTrigger(String key) {
+        OneSignal.getInAppMessages().removeTrigger(key);
+    }
+
+    @ReactMethod
+    public void removeTriggers(ReadableArray keys) {
+        OneSignal.getInAppMessages().removeTriggers(RNUtils.convertReableArrayIntoStringCollection(keys));
+    }
+
+    @ReactMethod
+    public void clearTriggers() {
+        OneSignal.getInAppMessages().clearTriggers();
+    }
+
+
+    // OneSignal.Location namespace methods
+    @ReactMethod
+    public void requestLocationPermission() {
+        OneSignal.getLocation().requestPermission(Continue.none());
+    }
+
+    @ReactMethod
+    public void isLocationShared(Promise promise) {
+        promise.resolve(OneSignal.getLocation().isShared());
+    }
+
+    @ReactMethod
+    public void setLocationShared(Boolean shared) {
+        OneSignal.getLocation().setShared(shared);
+    }
+
+
+    // OneSignal.Notifications namespace methods
+    @ReactMethod
+    public void setNotificationClickHandler() {
+        OneSignal.getNotifications().setNotificationClickHandler(new INotificationClickHandler() {
+            @Override
+            public void notificationClicked(INotificationClickResult result) {
+                try {
+                    sendEvent("OneSignal-notificationClicked", RNUtils.convertHashMapToWritableMap(RNUtils.convertNotificationClickResultToMap(result)));
+                } catch (JSONException e) {
+                    e.printStackTrace();
+                }
             }
-         }
-      });
-   }
-
-   @ReactMethod
-   public void promptForPushNotificationsWithUserResponse(final boolean fallbackToSettings, final Callback callback) {
-      final Callback[] callbackArr = new Callback[]{ callback };
-      OneSignal.promptForPushNotifications(fallbackToSettings, new OneSignal.PromptForPushNotificationPermissionResponseHandler() {
-         @Override
-         public void response(boolean accepted) {
-            if (callbackArr[0] != null) {
-               callbackArr[0].invoke(accepted);
-               callbackArr[0] = null;
+        });
+    }
+
+    @ReactMethod
+    public void setNotificationWillShowInForegroundHandler() {
+        OneSignal.getNotifications().setNotificationWillShowInForegroundHandler(new INotificationWillShowInForegroundHandler() {
+            @Override
+            public void notificationWillShowInForeground(INotificationReceivedEvent notificationReceivedEvent) {
+                try {
+                    INotification notification = notificationReceivedEvent.getNotification();
+                    String notificationId = notification.getNotificationId();
+                    notificationReceivedEventCache.put(notificationId, notificationReceivedEvent);
+                    sendEvent("OneSignal-notificationWillShowInForeground",
+                            RNUtils.convertHashMapToWritableMap(
+                                    RNUtils.convertNotificationToMap(notification)));
```

Comment:
I suspect you will need to wait here until `completeNotificationEvent` is called to drive the complete.  This is similar to what our [flutter SDK](https://github.com/OneSignal/OneSignal-Flutter-SDK/blob/user_model/main/android/src/main/java/com/onesignal/flutter/OneSignalNotifications.java#L170) does.

Reply:
Done